src/main/java/org/ice4j/ice/Agent.java

```diff
@@ -486,12 +488,14 @@ public Component createComponent(  IceMediaStream stream,
      * <tt>minPort</tt> and <tt>maxPort</tt> before reaching the maximum allowed
      * number of retries.
      */
-    public Component createComponent(  IceMediaStream stream,
-                                       Transport      transport,
-                                       int            preferredPort,
-                                       int            minPort,
-                                       int            maxPort,
-                                       KeepAliveStrategy keepAliveStrategy)
+    public Component createComponent(
+        IceMediaStream stream,
+        Transport transport,
+        int preferredPort,
+        int minPort,
+        int maxPort,
+        KeepAliveStrategy keepAliveStrategy,
+        boolean useComponentSocket)
```

Comment:
You might want to keep the existing constructor (with a default that resembles the previous behavior). Otherwise this needs a new API version.